app/frontend/src/features/communities/NewPageForm.tsx

```diff
@@ -22,53 +22,53 @@ type NewPageInputs = {
 };
 
 export default function CompleteSignup() {
-  const dispatch = useAppDispatch();
-
   const {
     control,
     register,
     handleSubmit,
     setValue,
     errors,
-    getValues,
   } = useForm<NewPageInputs>({
     shouldUnregister: false,
     mode: "onBlur",
   });
 
-  const [loading, setLoading] = useState(false);
-
-  const location = useLocation();
   const history = useHistory();
 
   const [alertState, setShowAlertState] = useState<
     "success" | "error" | undefined
   >();
   const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-  const submitForm = handleSubmit(async (data: NewPageInputs) => {
-    try {
-      const page = unwrapResult(await dispatch(createPage(data))) as Page.AsObject;
+  const {
+    mutate: createPage,
+    isLoading: isCreateLoading,
```

Comment:
if you expose `error` here too, you don't need the state for `errorMessage` and you can just remove the `onError` section